Sources/SpeziLLMOpenAI/LLMOpenAISession+Generation.swift

```diff
@@ -122,7 +125,14 @@ extension LLMOpenAISession {
             await MainActor.run {
                 context.append(functionCalls: functionCallContext)
             }
-            
+
+            let isFirstToolCall = self.incrementToolCallCounter(by: functionCalls.count)
+            if isFirstToolCall {
+                await MainActor.run {
+                    self.state = .callingTools
+                }
+            }
```

Comment:
I think it might make sense to make `decrementToolCallCounter` and do the state modification in there as well within the lock to ensure that we don't run into any race conditions? Applies to a few other parts of the code base and where we do that as well.